torchvision/io/video.py

```diff
@@ -0,0 +1,146 @@
+import av
+import gc
+import torch
+import numpy as np
+import math
+
+
+# PyAV has some reference cycles
+_CALLED_TIMES = 0
+_GC_COLLECTION_INTERVAL = 20
+
+
+def write_video(filename, video_array, fps):
+    """
+    Writes a 4d tensor in [T, H, W, C] format in a video file
+
+    Arguments:
+        filename (str): path where the video will be saved
+        video_array (Tensor[T, H, W, C]): tensor containing the individual frames,
+            as a uint8 tensor in [T, H, W, C] format
+        fps (Number): frames per second
+    """
+    video_array = torch.as_tensor(video_array, dtype=torch.uint8).numpy()
+
+    container = av.open(filename, mode='w')
+
+    stream = container.add_stream('mpeg4', rate=fps)
+    stream.width = video_array.shape[2]
+    stream.height = video_array.shape[1]
+    stream.pix_fmt = 'yuv420p'
+
+    for img in video_array:
+        frame = av.VideoFrame.from_ndarray(img, format='rgb24')
+        for packet in stream.encode(frame):
+            container.mux(packet)
+
+    # Flush stream
+    for packet in stream.encode():
+        container.mux(packet)
+
+    # Close the file
+    container.close()
+
+
+def _read_from_stream(container, start_offset, end_offset, stream, stream_name):
```

Comment:
Great question again.

I'm inclined to always return the audio at a fixed frequency (say 44kHz), so that the results are always consistent.

Thoughts?

Reply:
"I'm inclined to always return the audio at a fixed frequency"

definitely, BUT I think the exact value should be left for the user to decide? 
I feel like a simple if `sr != user_defined_sr call resampler` should be enough?

Reply:
I think that for a reading function, we should try to make it as simple as possible, and add additional transforms for resampling the audio / video if needed.

But you have great points about the stats that should be returned.

I'll modify the implementation to return a third argument, a dict with the fps etc.